desc/compute/_geometry.py

```diff
@@ -324,44 +386,68 @@ def _a(params, transforms, profiles, data, **kwargs):
     profiles=[],
     coordinates="",
     data=["R0", "a"],
+    parameterization=[
+        "desc.equilibrium.equilibrium.Equilibrium",
+        "desc.geometry.surface.FourierRZToroidalSurface",
+    ],
 )
 def _R0_over_a(params, transforms, profiles, data, **kwargs):
     data["R0/a"] = data["R0"] / data["a"]
     return data
 
 
 @register_compute_fun(
-    name="a_major/a_minor",
-    label="a_{\\mathrm{major}} / a_{\\mathrm{minor}}",
-    units="~",
-    units_long="None",
-    description="Elongation at a toroidal cross-section",
+    name="perimeter(z)",
+    label="P(\\zeta)",
+    units="m",
+    units_long="meters",
+    description="Perimeter of cross section as function of zeta",
     dim=1,
     params=[],
     transforms={"grid": []},
     profiles=[],
     coordinates="z",
-    data=["rho", "sqrt(g)", "g_tt", "R"],
+    data=["rho", "g_tt"],
+    parameterization=[
+        "desc.equilibrium.equilibrium.Equilibrium",
+        "desc.geometry.core.Surface",
+    ],
 )
-def _a_major_over_a_minor(params, transforms, profiles, data, **kwargs):
+def _perimeter_of_z(params, transforms, profiles, data, **kwargs):
     max_rho = jnp.max(data["rho"])
-    P = (  # perimeter at rho=1
+    data["perimeter(z)"] = (  # perimeter at rho ~ 1
         line_integrals(
             transforms["grid"],
             jnp.sqrt(data["g_tt"]),
             line_label="theta",
             fix_surface=("rho", max_rho),
-            expand_out=False,
+            expand_out=True,
         )
-        / max_rho
-    )
-    # surface area
-    A = surface_integrals(
-        transforms["grid"],
-        jnp.abs(data["sqrt(g)"] / data["R"]),
-        surface_label="zeta",
-        expand_out=False,
+        / max_rho  # to account for quadrature grid not having nodes out to rho=1
     )
+    return data
+
+
+@register_compute_fun(
+    name="a_major/a_minor",
+    label="a_{\\mathrm{major}} / a_{\\mathrm{minor}}",
+    units="~",
+    units_long="None",
+    description="Elongation at a toroidal cross-section",
+    dim=1,
+    params=[],
+    transforms={"grid": []},
+    profiles=[],
+    coordinates="z",
+    data=["A(z)", "perimeter(z)"],
+    parameterization=[
+        "desc.equilibrium.equilibrium.Equilibrium",
+        "desc.geometry.core.Surface",
+    ],
+)
+def _a_major_over_a_minor(params, transforms, profiles, data, **kwargs):
+    A = transforms["grid"].compress(data["A(z)"], surface_label="zeta")
+    P = transforms["grid"].compress(data["perimeter(z)"], surface_label="zeta")
     # derived from Ramanujan approximation for the perimeter of an ellipse
```

Comment:
it's always an approximation for any non-elliptical cross section, since the definitions of major and minor axes and elongation aren't uniquely defined in those cases. In practice it still gives reasonable results in all the cases we've tried it. For a triangular cross section you'd basically be getting the elongation of an equivalent elipse which would likely be close to 1, which is what we'd expect.